graphql/src/main/java/com/linecorp/armeria/server/graphql/GraphqlServiceBuilder.java

```diff
@@ -242,12 +244,22 @@ public GraphqlServiceBuilder errorHandler(GraphqlErrorHandler errorHandler) {
         return this;
     }
 
+    /**
+     * Sets the {@link ExecutionIdGenerator}.
+     * If not specified, {@link ExecutionIdGenerator#of()} is used by default.
+     */
+    public GraphqlServiceBuilder executionIdGenerator(ExecutionIdGenerator executionIdGenerator) {
+        this.executionIdGenerator = requireNonNull(executionIdGenerator, "executionIdGenerator");
+        return this;
+    }
+
     /**
      * Creates a {@link GraphqlService}.
      */
     public GraphqlService build() {
         final GraphQLSchema schema = buildSchema();
-        GraphQL.Builder builder = GraphQL.newGraphQL(schema);
+        GraphQL.Builder builder = GraphQL.newGraphQL(schema)
+                                         .executionIdProvider(executionIdGenerator.asExecutionProvider());
```

Comment:
note: double checked non-nullness is checked inside the `GraphQL` setter

Reply:
I think he just left a note to help future code reviewers. It's good as it is. 😄